dist_ir/ir/op.py

```diff
@@ -11,9 +11,11 @@
 @dataclass(frozen=True)
 class Op:
     op_type: str
-    name: str = ""
+    name: str = field(default="", hash=False, compare=False)
```

Comment:
This was an attempt to avoid having to compute string hash functions which seems to be the dominant source of overhead when I profile - unclear if this actually improved anything though

Reply:
I'd suggest we don't commit these changes in this PR if they aren't necessary for the results we reported in the submission. We can investigate this in a separate PR that does profiling/optimization of overhead. (Also, we should be careful that these changes, and the similar changes to the Value class, don't result in some unintended bugs elsewhere since they change the meaning of when two ops/values are equal.)